File: 卡码/代码随想录/1/07-回溯/C++/03-子集/LeetCode_0090.cpp
//90. 子集 II
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

class Solution {
	vector<int> path;
	vector<vector<int>> res;
public:
	void backtracking(vector<int>& nums,int index){
		res.push_back(path);
		if(index>=nums.size()) return;
		for(int i=index;i<nums.size();i++){
			if(i>index&&nums[i]==nums[i-1]) continue;
			path.push_back(nums[i]);
			backtracking(nums,i+1);
			path.pop_back();
		}
	}
	vector<vector<int>> subsetsWithDup(vector<int>& nums) {
		sort(nums.begin(),nums.end());
		backtracking(nums,0);
		return res;
	}
};

int main(){
  system("pause");
  return 0;
}

File: 卡码/代码随想录/1/07-回溯/Python/03-子集/LeetCode_0090.py
#90. 子集 II
from typing import *

class Solution:
  def __init__(self):
    self.path:List[int]=[]
    self.res:List[List[int]]=[]
  
  def backtracking(self,nums:List[int],index:int):
    self.res.append(self.path[:])
    for i in range(index,len(nums)):
      if i>index and nums[i]==nums[i-1]:
        continue
      self.path.append(nums[i])
      self.backtracking(nums,i+1)
      self.path.pop()
  
  def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
    nums.sort()
    self.backtracking(nums,0)
    return self.res